front/src/app/components/forms/agency/AddAgencyForm.tsx

```diff
@@ -120,6 +121,15 @@ const AgencyForm = ({ refersToOtherAgency }: AgencyFormProps) => {
       });
   };
 
+  const agenciesRetrieverMemoized = useMemo(
+    () => (departmentCode: DepartmentCode) =>
+      agencyGateway.getFilteredAgencies({
+        departmentCode,
+        kind: "withoutRefersToAgency",
+      }),
+    [],
+  );
```

Comment:
useCallback serait un peu plus court ?

```suggestion
  const agenciesRetrieverMemoized = useCallback(
    (departmentCode: DepartmentCode) =>
      agencyGateway.getFilteredAgencies({
        departmentCode,
        kind: "withoutRefersToAgency",
      }),
    [],
  );
```